service/history/task/cross_cluster_task.go

```diff
@@ -0,0 +1,309 @@
+		maxRetryCount)
+	return &crossClusterStartChildWorkflowTask{
+		crossClusterTask,
+	}
+}
+
+func newCrossClusterTask(
+	shard shard.Context,
+	taskInfo Info,
+	logger log.Logger,
+	timeSource clock.TimeSource,
+	maxRetryCount dynamicconfig.IntPropertyFn,
+) *crossClusterTaskBase {
+	var eventLogger eventLogger
+	if shard.GetConfig().EnableDebugMode &&
+		shard.GetConfig().EnableTaskInfoLogByDomainID(taskInfo.GetDomainID()) {
+		eventLogger = newEventLogger(logger, timeSource, defaultTaskEventLoggerSize)
+		eventLogger.AddEvent("Created task")
+	}
+
+	return &crossClusterTaskBase{
+		Info:          taskInfo,
+		shard:         shard,
+		priority:      ctask.NoPriority,
+		attempt:       0,
+		timeSource:    timeSource,
+		submitTime:    timeSource.Now(),
+		logger:        logger,
+		eventLogger:   eventLogger,
+		scopeIdx:      GetCrossClusterTaskMetricsScope(taskInfo.GetTaskType()),
+		scope:         nil,
+		maxRetryCount: maxRetryCount,
+	}
+}
+
+// Cross cluster signal workflow task
+
+func (c *crossClusterSignalWorkflowTask) Execute() error {
+	panic("Not implement")
+}
+
+func (c *crossClusterSignalWorkflowTask) Ack() {
+	panic("Not implement")
+
+}
+
+func (c *crossClusterSignalWorkflowTask) Nack() {
+	panic("Not implement")
+
+}
+
+func (c *crossClusterSignalWorkflowTask) HandleErr(
+	err error,
+) (retErr error) {
+	panic("Not implement")
+}
+
+func (c *crossClusterSignalWorkflowTask) RetryErr(
+	err error,
+) bool {
+	panic("Not implement")
+}
+
+func (c *crossClusterSignalWorkflowTask) IsReadyForPoll() bool {
+	panic("Not implement")
+}
+
+func (c *crossClusterSignalWorkflowTask) Update(interface{}) error {
+	panic("Not implement")
+}
+
+// Cross cluster cancel workflow task
+
+func (c *crossClusterCancelWorkflowTask) Execute() error {
+	panic("Not implement")
+}
+
+func (c *crossClusterCancelWorkflowTask) Ack() {
+	panic("Not implement")
+
+}
+
+func (c *crossClusterCancelWorkflowTask) Nack() {
+	panic("Not implement")
+
+}
+
+func (c *crossClusterCancelWorkflowTask) HandleErr(
+	err error,
+) (retErr error) {
+	panic("Not implement")
+}
+
+func (c *crossClusterCancelWorkflowTask) RetryErr(
+	err error,
+) bool {
+	panic("Not implement")
+}
+
+func (c *crossClusterCancelWorkflowTask) IsReadyForPoll() bool {
+	panic("Not implement")
+}
+
+func (c *crossClusterCancelWorkflowTask) Update(interface{}) error {
+	panic("Not implement")
+}
+
+// Cross cluster start child workflow task
+
+func (c *crossClusterStartChildWorkflowTask) Execute() error {
+	panic("Not implement")
+}
+
+func (c *crossClusterStartChildWorkflowTask) Ack() {
+	panic("Not implement")
+
+}
+
+func (c *crossClusterStartChildWorkflowTask) Nack() {
+	panic("Not implement")
+
+}
+
+func (c *crossClusterStartChildWorkflowTask) HandleErr(
+	err error,
+) (retErr error) {
+	panic("Not implement")
+}
+
+func (c *crossClusterStartChildWorkflowTask) RetryErr(
+	err error,
+) bool {
+	panic("Not implement")
+}
+
+func (c *crossClusterStartChildWorkflowTask) IsReadyForPoll() bool {
```

Comment:
Since we are using shared processing state right now, looks like this method can also be moved to crossClusterTaskBase impl?

Reply:
Yes. Will mmove to crossClusterTaskBase